lib/adagrams.rb

```diff
@@ -0,0 +1,113 @@
+def draw_letters
+  letter_distribution = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1
+  }
+  letter_pool = []
+  letter_distribution.each do |letter, number|
+    number.times do
+      letter_pool << letter.to_s
+    end
+  end
+  letters_in_hand = letter_pool.sample(10)
+  return letters_in_hand
+end
+
+def uses_available_letters?(input, letters_in_hand)
+  hand_copy = letters_in_hand.clone
+  input_array = input.chars
+  input_array.each do |char|
+    index = hand_copy.index(char)
+    if !index || !hand_copy.delete_at(index)
+      return false
+    end
+  end
+  return true
+end
+
+# 3
+def score_word(word)
+  letter_array = word.upcase.split(//)
+  word_score = 0
+  letter_array.each do |each_letter|
+    case each_letter
```

Comment:
While this `case` statement works, it means that the information about which letter has which score is locked into this piece of code, and can't easily be used elsewhere. For example, if you wanted to display the value of each letter in a hand, you would need to repeat this work.

An alternative approach would be to store the letter scores in a hash, something like this:

```ruby
LETTER_SCORES = {
  "A" => 1
  "B" => 3,
  "C" => 3,
  "D" => 2,
  # ...
}